build.rs

```diff
@@ -1,103 +1,53 @@
 use bindgen;
-#[cfg(all(
-    not(feature = "pkg_config"),
-    feature = "bundled_proj",
-    not(feature = "nobuild")
-))]
 use cmake;
-#[cfg(all(
-    not(feature = "pkg_config"),
-    feature = "bundled_proj",
-    not(feature = "nobuild")
-))]
 use flate2::read::GzDecoder;
-#[cfg(all(
-    not(feature = "pkg_config"),
-    feature = "bundled_proj",
-    not(feature = "nobuild")
-))]
 use std::fs::File;
 
-#[cfg(all(
-    feature = "pkg_config",
-    not(feature = "bundled_proj"),
-    not(feature = "nobuild")
-))]
 use pkg_config;
 use std::env;
 use std::path::PathBuf;
-#[cfg(all(
-    not(feature = "pkg_config"),
-    feature = "bundled_proj",
-    not(feature = "nobuild")
-))]
 use tar::Archive;
 
-#[cfg(all(
-    feature = "pkg_config",
-    not(feature = "bundled_proj"),
-    not(feature = "nobuild")
-))]
 const MINIMUM_PROJ_VERSION: &str = "7.1.0";
 
 #[cfg(feature = "nobuild")]
 fn main() {} // Skip the build script on docs.rs
 
-// We sometimes need additional search paths, which we get using pkg-config
-#[cfg(all(
-    feature = "pkg_config",
-    not(feature = "nobuild"),
-    not(feature = "bundled_proj")
-))]
-fn main() {
-    let pk = pkg_config::Config::new()
+#[cfg(not(feature = "nobuild"))]
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let include_path = if cfg!(feature = "bundled_proj") {
+        println!("feature flags specified source build");
+        build_from_source()?
+    } else {
+        pkg_config::Config::new()
         .atleast_version(MINIMUM_PROJ_VERSION)
         .probe("proj")
-        .expect(&format!(
-            "Your PROJ version may be too old. You need at least version {}",
-            MINIMUM_PROJ_VERSION
-        ));
-    // Tell cargo to tell rustc to link the system proj
-    // shared library.
-    println!("cargo:rustc-link-search=native={:?}", pk.link_paths[0]);
-    println!("cargo:rustc-link-lib=proj");
-    let include_path = pk.include_paths[0].to_string_lossy();
-
-    // The bindgen::Builder is the main entry point
-    // to bindgen, and lets you build up options for
-    // the resulting bindings.
-    let bindings = bindgen::Builder::default()
-        .clang_arg(format!("-I{}", include_path))
-        .trust_clang_mangling(false)
-        .blacklist_type("max_align_t")
-        // The input header we would like to generate
-        // bindings for.
-        .header("wrapper.h")
-        // Finish the builder and generate the bindings.
-        .generate()
-        // Unwrap the Result and panic on failure.
-        .expect("Unable to generate bindings");
+        .and_then(|pk| {
+            println!("found acceptable libproj already installed at: {:?}", pk.link_paths[0]);
```

Comment:
Should this be `eprintln!` to not mess with stdout that cargo uses to pass options to rustc, etc.?